File: dx-api/app/consts/score_rating.go
package consts

// ScoreRating holds the label for a score accuracy range.
type ScoreRating struct {
	Label string
}

// scoreRatingEntry is used internally for lookup.
type scoreRatingEntry struct {
	MinAccuracy float64
	Label       string
}

// scoreRatings defines rating tiers in descending order of accuracy.
var scoreRatings = []scoreRatingEntry{
	{MinAccuracy: 0.9, Label: "优秀"},
	{MinAccuracy: 0.7, Label: "良好"},
	{MinAccuracy: 0.6, Label: "及格"},
	{MinAccuracy: 0, Label: "继续加油"},
}

// GetScoreRating returns the score rating label based on accuracy (0-1).
func GetScoreRating(accuracy float64) ScoreRating {
	for _, r := range scoreRatings {
		if accuracy >= r.MinAccuracy {
			return ScoreRating{Label: r.Label}
		}
	}
	return ScoreRating{Label: "继续加油"}
}

File: dx-web/src/consts/score-rating.ts
export type ScoreRating = {
  label: string;
  colorClass: string;
  bgClass: string;
};

export const SCORE_RATINGS = [
  { minAccuracy: 0.9, label: "优秀", colorClass: "text-teal-600", bgClass: "bg-teal-50" },
  { minAccuracy: 0.7, label: "良好", colorClass: "text-blue-600", bgClass: "bg-blue-50" },
  { minAccuracy: 0.6, label: "及格", colorClass: "text-amber-600", bgClass: "bg-amber-50" },
  { minAccuracy: 0, label: "继续加油", colorClass: "text-rose-500", bgClass: "bg-rose-50" },
] as const;

/** Get score rating label and color based on accuracy (0-1) */
export function getScoreRating(accuracy: number): ScoreRating {
  const match = SCORE_RATINGS.find((r) => accuracy >= r.minAccuracy);
  return match ?? { label: "继续加油", colorClass: "text-rose-500", bgClass: "bg-rose-50" };
}
